pkg/controllers/provisioning/provisioner.go

```diff
@@ -359,6 +359,8 @@ func (p *Provisioner) Create(ctx context.Context, n *scheduler.NodeClaim, opts .
 		metrics.ReasonLabel:   options.Reason,
 		metrics.NodePoolLabel: nodeClaim.Labels[v1beta1.NodePoolLabelKey],
 	}).Inc()
+	// Update the nodeclaim manually in state to avoid evenutal consistency delay races with our watcher
```

Comment:
I'd personally expound on this comment a bit more. I can imagine quickly forgetting the context on why we added this and this seems like a pretty critical line to ensure that we aren't forging ahead without knowing that we just created something